src/main/kotlin/io/github/raeperd/realworldspringbootkotlin/domain/ArticleFavoriteService.kt

```diff
@@ -0,0 +1,31 @@
+package io.github.raeperd.realworldspringbootkotlin.domain
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+
+@Transactional
+@Service
+class ArticleFavoriteService(
+    private val userRepository: ReadOnlyUserRepository,
+    private val articleRepository: ArticleRepository
+) {
+    fun favoriteArticle(userId: Long, slug: String): ArticleDTO {
+        val user = userRepository.findUserByIdOrThrow(userId)
+        val article = articleRepository.findArticleBySlugOrThrow(slug)
+        user.favoriteArticle(article)
+        return articleRepository.saveArticle(article)
+            .toArticleDTO(user)
+    }
+
+    fun unfavoriteArticle(userId: Long, slug: String): ArticleDTO {
+        val user = userRepository.findUserByIdOrThrow(userId)
+        val article = articleRepository.findArticleBySlugOrThrow(slug)
+        if (!article.isFavoritedByUser(user)) {
```

Comment:
This branch is useless

Reply:
Resolved by [d0f1ef7](https://github.com/raeperd/realworld-springboot-kotlin/pull/49/commits/d0f1ef7903a3db45514a383f98ccabb4dc39a0f9)